sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py

```diff
@@ -0,0 +1,299 @@
+                    )
+                # materialization identity = identity when created through feature store operations
+                if (
+                    offline_Store_connection.name == OFFLINE_STORE_CONNECTION_NAME
+                    and feature_store.identity
+                    and feature_store.identity.user_assigned_identities
+                    and isinstance(feature_store.identity.user_assigned_identities[0], ManagedIdentityConfiguration)
+                ):
+                    feature_store.materialization_identity = feature_store.identity.user_assigned_identities[0]
+
+        return feature_store
+
+    # @monitor_with_activity(logger, "FeatureStore.BeginCreate", ActivityType.PUBLICAPI)
+    @distributed_trace
+    # pylint: disable=arguments-differ
+    def begin_create(
+        self,
+        feature_store: FeatureStore,
+        update_dependent_resources: bool = False,
+        **kwargs: Dict,
+    ) -> LROPoller[FeatureStore]:
+        """Create a new FeatureStore.
+
+        Returns the feature store if already exists.
+
+        :param feature store: FeatureStore definition.
+        :type feature store: FeatureStore
+        :type update_dependent_resources: boolean
+        :return: An instance of LROPoller that returns a FeatureStore.
+        :rtype: ~azure.core.polling.LROPoller[~azure.ai.ml.entities.FeatureStore]
+        """
+        if feature_store.offline_store and feature_store.offline_store.type != OFFLINE_MATERIALIZATION_STORE_TYPE:
+            raise ValidationError("offline store type should be azure_data_lake_gen2")
+        if feature_store.offline_store and not feature_store.materialization_identity:
+            raise ValidationError("materialization_identity is required to setup offline store")
+
+        def get_callback():
+            return self.get(feature_store.name)
+
+        return super().begin_create(
+            workspace=feature_store,
+            update_dependent_resources=update_dependent_resources,
+            get_callback=get_callback,
+            offline_store_target=feature_store.offline_store.target if feature_store.offline_store else None,
+            materialization_identity=feature_store.materialization_identity,
+            **kwargs,
+        )
+
+    # @monitor_with_activity(logger, "FeatureStore.BeginUpdate", ActivityType.PUBLICAPI)
+    @distributed_trace
+    # pylint: disable=arguments-renamed
+    def begin_update(
+        self,
+        feature_store: FeatureStore,
+        *,
+        update_dependent_resources: bool = False,
+        **kwargs: Dict,
+    ) -> LROPoller[FeatureStore]:
+        """Update friendly name, description, materialization identities or tags of a feature store.
+
+        :param feature store: FeatureStore resource.
+        :param update_dependent_resources: gives your consent to update the feature store dependent resources.
```

Comment:
is this from spec & general pattern across other operations ?

Reply:
yes, this is from workspace operations